contracts/protocol/v1/proxies/P1SoloBridgeProxy.sol

```diff
@@ -77,24 +77,28 @@ contract P1SoloBridgeProxy is
         "address perpetual,",
         "uint256 soloAccountNumber,",
         "uint256 soloMarketId,",
-        "bool toPerpetual,",
         "uint256 amount,",
-        "uint256 expiration,",
-        "bytes32 salt",
+        "bytes32 options",
         ")"
     ));
 
+    // Constants for the options field of the Transfer struct.
+    bytes32 private constant OPTIONS_MASK_TRANSFER_MODE = bytes32(uint256((1 << 8) - 1));
+    bytes32 private constant OPTIONS_MASK_EXPIRATION = bytes32(uint256((1 << 120) - 1));
+    uint256 private constant OPTIONS_OFFSET_EXPIRATION = 8;
+    bytes32 private constant TRANSFER_MODE_SOME_TO_PERPETUAL = bytes32(uint256(0));
+    bytes32 private constant TRANSFER_MODE_SOME_TO_SOLO = bytes32(uint256(1));
+    bytes32 private constant TRANSFER_MODE_ALL_TO_PERPETUAL = bytes32(uint256(2));
```

Comment:
it might be better to make this an `enum` type in solidity (automatically throws error if non-supported enum value)